zkevm-circuits/src/evm_circuit/util/math_gadget/rlp.rs

```diff
@@ -67,10 +72,19 @@ impl<F: Field> RlpU64Gadget<F> {
             });
         }
 
-        let is_lt_128 = cb.query_bool();
-        cb.condition(is_lt_128.expr(), |cb| {
-            cb.range_lookup(value, 128);
-        });
+        // If is_lt_128, then value < 128, checked by a lookup.
+
+        // Otherwise, then value >= 128, checked as follows:
+        // - Either the first byte is not the most significant, and there is a more significant one;
+        // - Or the first byte is the most significant, and it is >= 128. value ∈ [128, 256) (value
+        //   - 128) ∈ [0, 128)
+        let byte_128 = value_rlc.cells[0].expr() - 128.expr();
+        let is_first = is_most_significant_byte[0].expr();
+        cb.require_boolean("is_first", is_first.clone());
```

Comment:
This seems unnecessary since all `is_most_significant_byte` are already bool when they are queried.
```suggestion
```